src/Transaction/TransactionBuilder.php

```diff
@@ -265,10 +265,11 @@ public function getTransaction()
                 $script = $this->transaction->getInputs()->getInput($i)->getScript();
             }
 
-            $transaction->getInputs()->getInput($i)->setScript($script);
+            $input = $transaction->getInputs()->getInput($i);
+            $transaction->getInputs()->setInput($i, new TransactionInput($input->getTransactionId(), $input->getVout(), $script, $input->getSequence()));
```

Comment:
`setInput` == replace, I guess renaming to `replaceInput` and making it check that it can only replace and not add would be a good idea.